docs/stats.md

```diff
@@ -693,6 +695,273 @@ framework which has the same calling conventions as the single site stats,
+or a pair of vectors for the row sites and column sites separately.
+
+The following example produces a matrix containing {math}`r^2` computed pairwise
+between the first 4 sites in the tree sequence (all samples included). In our
+computations, row sites are used as the lefthand locus and column sites are used
+as the righthand locus (though in most cases, the LD matrix will be symmetric).
+
+```{code-cell} ipython3
+ld = ts.ld_matrix(sites=[[0, 1, 2, 3]])
+print(ld)
+```
+
+Here's an example demonstrating how we can specify the row and column sites
+independently of each other. We're specifying 4 columns and 1 row, selecting the
+first row from the matrix shown above.
+```{code-cell} ipython3
+ld = ts.ld_matrix(sites=[[0], [0, 1, 2, 3]])
+print(ld)
+```
+
+Because we implement two-locus statistics for multi-allelic data, we need a
+method for combining the statistical results from each allele into one summary
+for a pair of sites. We have two methods for combining results from multiple
+alleles:
+
+`hap_weighted`
+: {math}`\sum_{i=1}^{n}\sum_{j=1}^{m}p(A_{i}B_{j})f_{ij}`
+
+  where {math}`f_{ij}` is the result of computing the summary function on data
+  from the {math}`i`th row site and the {math}`j`th column site and
+  {math}`p(A_{i}B_{j})` is the frequency of haplotype {math}`A_{i}B_{j}`. This
+  method was first introduced in
+  [Karlin (1981)](https://doi.org/10.1111/j.1469-1809.1981.tb00308.x), and was
+  reviewed in [Zhao (2007)](https://doi.org/10.1017/S0016672307008634).
+
+  This method multiplies each result by the frequency of the haplotype under
+  consideration (the {math}`[i,j]` alleles for a given pair of sites), then sums
+  the results to obtain the statistic for the two focal sites.
+
+`total_weighted`
+: {math}`\frac{1}{(n-\mathbb{1}_{p}) (m-\mathbb{1}_{p})}\sum_{i=1}^{n}\sum_{j=1}^{m}f_{ij}`
+
+  where {math}`\mathbb{1}_{p}` is an indicator function conditioned on whether
+  or not our statistic is polarized, {math}`n` is the number of alleles in site
+  {math}`a`, and {math}`m` is the number of alleles in site {math}`b`.
+
+  This method assigns equal weight to each of the possible haplotypes for a
+  given pair of sites, it might be more aptly named "uniform_weighted". In
+  essence, we are taking the arithmetic mean of each statistic.
+
+Out of all of the available summary functions, only {math}`r^2` uses
+`hap_weighted` normalisation. The rest behave correctly under a uniform weighting
+(`total_weighted`).
+
+(sec_stats_two_locus_branch)=
+
+#### Branch
+
+The `branch` mode computes two-locus statistics between pairs of trees, the
+resulting statistic is a summary of the branch lengths between both trees, over
+the area formed by multiplying the spans of the two trees. To perform this
+computation, we use the partition formed by each node in the tree to generate
+haplotype counts, then we sum each pairwise comparison weighted by the branch
+length above the node. The time complexity of this method is quadratic, due to
```

Comment:
It may be useful to provide some interpretation here, that (similar to the single-site statistics) the branch statistic is proportional to the expected statistic under an infinite sites model (with mutation rate 1), conditioned on the pair of trees. Or something like that.